pkg/cloudprovider/cloudprovider.go

```diff
@@ -112,6 +115,59 @@ func withInspection() autorest.PrepareDecorator {
 	}
 }
 
+// CheckUserMSI checks if the identity has been assigned to the node
+func (c *Client) CheckUserMSI(userAssignedMSIID string, node *corev1.Node) bool {
+	idH, _, err := c.getIdentityResource(node)
```

Comment:
Arent' these calls costly ? in the sense that they have to ARM every time ?

Reply:
@kkmsft You're right! we can avoid the multiple calls to ARM by just getting the list once and then validating existence/non-existence of all assigned ids based on that list.